services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/DpmServiceImplV2.java

```diff
@@ -119,13 +124,26 @@ private UnicastGoalStateV2 buildUnicastGoalState(NetworkConfiguration networkCon
             portService.buildPortState(networkConfig, portEntities, unicastGoalState);
         }
 
+        Map<String, InternalSubnetPorts> internalSubnetPorts = subnetPortsCache.getSubnetPorts(networkConfig);
+        Map<String, PortHostInfo> portHostInfoMap = portHostInfoCache.getPortHostInfo(networkConfig);
+        synchronized (this) {
+            try(Transaction tx = subnetPortsCache.getTransaction().start()) {
+                subnetPortsCache.updateSubnetPorts(internalSubnetPorts);
+                portHostInfoCache.updatePortHostInfo(portHostInfoMap);
+                tx.commit();
+            }
+        }
+
         vpcService.buildVpcStates(networkConfig, unicastGoalState);
-        subnetService.buildSubnetStates(networkConfig, unicastGoalState, multicastGoalState);
-        neighborService.buildNeighborStates(networkConfig, hostIp, unicastGoalState, multicastGoalState);
+        subnetService.buildSubnetStates(networkConfig, unicastGoalState);
         securityGroupService.buildSecurityGroupStates(networkConfig, unicastGoalState);
         dhcpService.buildDhcpStates(networkConfig, unicastGoalState);
-        routerService.buildRouterStates(networkConfig, unicastGoalState, multicastGoalState);
-        patchGoalstateForNeighbor(networkConfig, unicastGoalState);
```

Comment:
Oh we don't need to patch GS any more? good. will it break any scenario? @cj-chung

Reply:
`patchGoalstate` related codes are trying to patch the missing neighbor info for L3 neighbor, since we have pushed the neighbor calculation into DPM by looking up caches, so we don't need the patch anymore. However, we better test out the scenarios with the new update.

Reply:
It tested on Kolla and Medina.